tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+        self.log.info("Verify Create bucket: %s with access key: %s and secret key: %s",
+                      self.src_bkt, self.akey, self.skey)
+        bucket_created = s3_misc.create_bucket(self.src_bkt, self.akey, self.skey)
+        assert bucket_created, "Failed to create bucket"
+        self.buckets_created.append([self.src_bkt, self.akey, self.skey])
+        self.log.info("ENDED: Test setup.")
+
+    def teardown_method(self):
+        """
+        Teardown will be invoked after running each test case.
+
+        It will clean all resources which are getting created during
+        test execution such as S3 buckets and the objects present into that bucket.
+        """
+        self.log.info("STARTED: Test teardown.")
+        self.log.info("Deleting buckets %s & associated objects", self.buckets_created)
+        buckets_deleted = []
+        iam_deleted = []
+        for bucket in self.buckets_created:
+            resp = s3_misc.delete_objects_bucket(bucket[0], bucket[1], bucket[2])
+            if resp:
+                buckets_deleted.append(bucket)
+            else:
+                self.log.error("Bucket deletion failed for %s ", bucket)
+        self.log.info("buckets deleted %s", buckets_deleted)
+        for bucket in buckets_deleted:
+            self.buckets_created.remove(bucket)
+
+        self.log.info("Deleting iam account %s created in test", self.created_iam_users)
+        for iam_user in self.created_iam_users:
+            resp = self.csm_obj.delete_iam_user(iam_user)
+            if resp.status_code == HTTPStatus.OK:
+                iam_deleted.append(iam_user)
+            else:
+                self.log.error("IAM deletion failed for %s ", iam_user)
+        self.log.info("IAMs deleted %s", iam_deleted)
+        for iam in iam_deleted:
+            self.created_iam_users.remove(iam)
+        assert_utils.assert_true(len(self.buckets_created) == 0, "Bucket deletion failed")
+        assert_utils.assert_true(len(self.created_iam_users) == 0, "IAM deletion failed")
+        self.log.info("ENDED: Test teardown.")
+
+    def create_iam_user(self):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        self.log.info("Creating S3 account and IAM user")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        user_id = resp.json()["keys"][0]['user']
+        self.created_iam_users.add(user_id)
+        resp1 = self.csm_obj.compare_iam_payload_response(resp, payload)
+        self.log.info("Printing response %s", resp1)
+        assert resp1[0], resp1[1]
+        akey = resp.json()["keys"][0]["access_key"]
+        skey = resp.json()["keys"][0]["secret_key"]
+        return akey, skey, user_id
+
+    def set_get_user_quota(self, config_dict, user_id):
```

Comment:
Let's move this to s3_misc

Reply:
As explained above, cannot move to s3misc.

Reply:
This approach will cause duplication, we cannot have helper code isolated in test modules that is surely to be needed in future tests. s3_misc has a lot of code which interacts with CSM and seems to be the best fit as of now. If not we can discuss the alternative lib locations this should be moved to.

Reply:
I disagree as S3misc is purely defined as a replacement of s3 test library function.
There are already common APIs defined for set and get quota in library code. For my test purpose I am just trying to club 2-3 common used functions in one wrapper. 
It is same as we write common function specific to that file in the same file.

Reply:
I do not see an issue with s3_misc. In fact it looks like a good location to have cross component interaction libs for S3. But if you read my comment carefully, I have clearly mentioned - "If not we can discuss the alternative lib locations this should be moved to."

Reply:
These wrappers are specific to tests only, combined all common code from library required to simplify test purpose.

Reply:
This is where we can utilize the common libs layer, as well. As we are discussing this offline, let's put a pause to this discussion here, find some middle ground in an offline discussion and then resolve this once we have come to an agreement.